client/network/src/service.rs

```diff
@@ -942,87 +910,129 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		if peer_id == self.local_peer_id {
 			return Err("Local peer ID cannot be added as a reserved peer.".to_string())
 		}
-		self.peerset.add_reserved_peer(peer_id.clone());
+
+		let _ = self
+			.to_worker
+			.unbounded_send(ServiceToWorkerMsg::AddReserved(peer_id.clone()));
 		let _ = self
 			.to_worker
 			.unbounded_send(ServiceToWorkerMsg::AddKnownAddress(peer_id, addr));
 		Ok(())
 	}
 
-	/// Configure an explicit fork sync request.
-	/// Note that this function should not be used for recent blocks.
-	/// Sync should be able to download all the recent forks normally.
-	/// `set_sync_fork_request` should only be used if external code detects that there's
-	/// a stale fork missing.
-	/// Passing empty `peers` set effectively removes the sync request.
-	pub fn set_sync_fork_request(&self, peers: Vec<PeerId>, hash: B::Hash, number: NumberFor<B>) {
+	/// Removes a `PeerId` from the list of reserved peers.
+	pub fn remove_reserved_peer(&self, peer_id: PeerId) {
 		let _ = self
 			.to_worker
-			.unbounded_send(ServiceToWorkerMsg::SyncFork(peers, hash, number));
+			.unbounded_send(ServiceToWorkerMsg::RemoveReserved(peer_id));
 	}
 
-	/// Modify a peerset priority group.
+	/// Add peers to a peerset priority group.
 	///
-	/// Each `Multiaddr` must end with a `/p2p/` component containing the `PeerId`.
+	/// Each `Multiaddr` must end with a `/p2p/` component containing the `PeerId`. It can also
+	/// consist of only `/p2p/<peerid>`.
 	///
 	/// Returns an `Err` if one of the given addresses is invalid or contains an
 	/// invalid peer ID (which includes the local peer ID).
-	//
-	// NOTE: even though this function is currently sync, it's marked as async for
-	// future-proofing, see https://github.com/paritytech/substrate/pull/7247#discussion_r502263451.
-	pub async fn set_priority_group(&self, group_id: String, peers: HashSet<Multiaddr>) -> Result<(), String> {
+	pub fn add_peers_set_reserved(&self, protocol: Cow<'static, str>, peers: HashSet<Multiaddr>) -> Result<(), String> {
```

Comment:
This wasn't possible anymore due to the need to integrate them [here](https://github.com/paritytech/substrate/blob/0e91d3d5d340058726bf1230e69d755a793fd7cf/client/network-gossip/src/lib.rs#L106-L122).
Making this `Network` trait an async trait also isn't possible because it's called from [a `poll` function](https://github.com/paritytech/substrate/blob/0e91d3d5d340058726bf1230e69d755a793fd7cf/client/network-gossip/src/bridge.rs#L182). I didn't have the courage to rewrite this `Future` as an async function.

Reply:
I believe the correct method name here would be `add_peer_set_reserved`; "a peer set" –> "many peer sets".